cpp/src/arrow/extension/tensor_array.cc

```diff
@@ -0,0 +1,103 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/extension/tensor_array.h"
+#include "arrow/tensor.h"
+
+namespace arrow {
+namespace extension {
+
+static std::string get_name(const std::shared_ptr<DataType>& type,
+                            const std::vector<int64_t>& shape,
+                            const std::vector<int64_t>& strides) {
+  std::stringstream s;
+  s << "arrow.extension.tensor<type=" << *type << ", shape=(";
+  for (uint64_t i = 0; i < shape.size(); i++) {
+    s << shape[i];
+    if (i < shape.size() - 1) {
+      s << ", ";
+    }
+  }
+  s << "), strides=(";
+  for (uint64_t i = 0; i < strides.size(); i++) {
+    s << strides[i];
+    if (i < strides.size() - 1) {
+      s << ", ";
+    }
+  }
+  s << ")>";
+  return s.str();
+}
+
+std::string TensorArrayType::extension_name() const {
```

Comment:
Extension name should be the same for all instances of `TensorArrayType`.
(the reason it can't be a `static` method is that it has to be `virtual`; unfortunately C++ doesn't allow both)

Reply:
Changed to `"arrow.fixed_size_tensor"`.